flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/sink/HBaseSinkFunction.java

```diff
@@ -213,6 +225,7 @@ public void close() throws Exception {
 
         if (mutator != null) {
             try {
+                flush();
```

Comment:
While your thread safety looks ok (besides the `static` `Map`), generally speaking `flush()` on `close()` can cause issues. If the destination is down, the job might fail to stop. A better solution is to checkpoint the internal buffer. Longer term we can consider migrating to the Async Sink base.

Reply:
Hi @dannycranmer , personally I reckon this might not be an issue because according to the documentation, the `close()` method of mutator inherently flushes buffered data to the HBase server before closing the connection, so the flush logic is already there before this PR. 

Also, as it is already in the `try ... catch` block, when an `IOException` is thrown by the client during the flush, the job-stopping process would not be interrupted as well.

Reply:
master:
close(){
...
mutator.close();
...
}

this pr:
close(){
...
flush();
mutator.close();
...
}

On the one hand, mutator.close will call mutator.flush as well. I think the metioned problem will happen at master as well. On the other hand, If the destination is down, mutator.close/mutator.flush will throw IOException.